arc.m

```diff
@@ -14,9 +14,36 @@
 
 #ifndef NO_PTHREADS
 #include <pthread.h>
-pthread_key_t ARCThreadKey;
+typedef pthread_key_t arc_tls_key_t;
+#define arc_tls_store pthread_setspecific
+#define arc_tls_load pthread_getspecific
+#define TLS_CALLBACK(name) void name
+
+static inline pthread_key_t arc_tls_key_create(void(*cleanupFunction)(void*))
+{
+	pthread_key_t key;
+	pthread_key_create(&key, cleanupFunction);
+	return key;
+}
+#else // if defined(NO_PTHREADS)
```

Comment:
This should be in some kind of `#ifdef` Windows thing - Windows isn't the only platform with no pthreads.  I suspect that for Windows this actually shouldn't be in the `#else` bit at all: even if we have Pthreads on Windows via the MinGW pthreads implementation, we'd probably prefer to use the native Windows APIs.